src/JsonRpcProxy/Adaptor.js

```diff
@@ -0,0 +1,266 @@
+        req.params[0] = format.formatAddress(req.params[0], proxy.networkId);
+        format.formatEpochOfParams(req.params, 1);
+        await next();
+    }
+
+    async function getTransactionReceipt(req, res, next) {
+        req.method = defaultMethodAdaptor(req.method, req.params);
+        await next();
+        if (!res || !res.result) return;
+        let txReceipt = res.result;
+        if (txReceipt.contractCreated) {
+            txReceipt.contractCreated = format.formatHexAddress(txReceipt.contractCreated);
+        }
+        txReceipt.from = format.formatHexAddress(txReceipt.from);
+        txReceipt.to = format.formatHexAddress(txReceipt.to);
+        if (txReceipt.logs) {
+            txReceipt.logs.forEach(
+            l => (l.address = format.formatHexAddress(l.address))
+            );
+        }
+
+        txReceipt.contractAddress = txReceipt.contractCreated;
+        txReceipt.blockNumber = txReceipt.epochNumber;
+        txReceipt.transactionIndex = txReceipt.index;
+        txReceipt.status = Number.parseInt(txReceipt.outcomeStatus)
+            ? "0x0"
+            : "0x1"; // conflux and eth status code is opposite
+        txReceipt.cumulativeGasUsed = txReceipt.gasUsed; // NOTE: this is an fake value
+        util.delKeys(txReceipt, [
+            "contractCreated",
+            "epochNumber",
+            "index",
+            "outcomeStatus",
+            "stateRoot"
+        ]);
+        return res;
+    }
+
+    async function sendTransaction(req, res, next) {
+        req.method = methodAdaptor(req.method, req.params);
+        if (params.length === 0) throw new Error('The first parameter should be an transaction');
+        req.params[0] = await format.formatTxParams(proxy.cfx, req.params[0]);
+        if (proxy.cfx.wallet.has(req.params[0].from)) {
+            let tx = new Transaction(req.params[0]);
+            let account = proxy.cfx.wallet.get(req.params[0].from);
+            tx.sign(account.privateKey, proxy.networkId);
+            req.params[0] = tx.serialize();
+        }
+        await next();
+    }
+    
+    async function getNetVersion(req, res, next) {
+        req.method = defaultMethodAdaptor(req.method, req.params);
+        await next();
+        if(!res || !res.result) return;
+        res.result = res.result.networkId;
+    }
+
+    async function getBalance(req, res, next) {
+        req.method = defaultMethodAdaptor(req.method, req.params);
+        req.params[0] = format.formatAddress(req.params[0], proxy.networkId);
+        format.formatEpochOfParams(req.params, 1);
+        await next();
+    }
+
+    async function estimateGas(req, res, next, end) {
+        req.method = defaultMethodAdaptor(req.method, req.params);
+        req.params = format.formatCommonInput(req.params, proxy.networkId);
+        await next();
+        if(!res || !res.result) return;
+        res.result = res.result.gasUsed;
+    }
+
+    function methodAdaptor(method, params) {
```

Comment:
This method's name is a little confusing with above adaptMethod

Reply:
renamed